File: src/portfolio_manager/models/deposit.py
"""Deposit model."""

from dataclasses import dataclass
from datetime import date, datetime
from decimal import Decimal
from typing import Optional
from uuid import UUID


@dataclass
class Deposit:
    """A deposit transaction."""

    id: UUID
    account_id: UUID
    amount: Decimal
    deposit_date: date
    created_at: datetime
    updated_at: datetime
    note: Optional[str] = None

File: internal/models/deposit.go
package models

import (
	"database/sql"

	"github.com/kadragon/portfolio-manager/internal/datex"
	"github.com/kadragon/portfolio-manager/internal/ktime"
	"github.com/kadragon/portfolio-manager/internal/numeric"
	"github.com/kadragon/portfolio-manager/internal/uuidx"
)

type Deposit struct {
	ID          uuidx.UUID
	Amount      numeric.Decimal
	DepositDate datex.Date
	Note        sql.NullString
	CreatedAt   ktime.Time
	UpdatedAt   ktime.Time
}
